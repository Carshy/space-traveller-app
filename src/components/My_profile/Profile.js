import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import './profile.css';
import { PopulateMissionProfile } from '../../redux/mission/mission';
import { PopulateRocketsProfile } from '../../redux/rockets/rocket';

const Profile = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions);
  const rockets = useSelector((state) => state.rockets);

  useEffect(() => {
    dispatch(PopulateMissionProfile());
    dispatch(PopulateRocketsProfile());
  }, []);

  return (
    <div className="myProfile">
      <section className="my-comp">
        <h2>My Mission</h2>
        <ul className="reserved-container">
          { missions.map((mission) => (mission.reserved === true ? (
            <li className="reserved-item" key={mission.mission_id}>
              {' '}
              {mission.mission_name}
              {' '}
            </li>
          ) : null))}
        </ul>
      </section>

      <section className="my-comp">
        <h2>My Rockets</h2>
        <ul className="reserved-container">
          { rockets.map((rocket) => (rocket.reserved === true ? (
            <li className="reserved-item" key={rocket.id}>
              {' '}
              {rocket.rocket_name}
              {' '}
            </li>
          ) : null))}
        </ul>
      </section>
    </div>
  );
};

export default Profile;
