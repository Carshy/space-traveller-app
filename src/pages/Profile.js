import React from 'react';

function Profile() {
  return (
    <div>
      <h1>Our Profile</h1>
    </div>
  );
}

export default Profile;
