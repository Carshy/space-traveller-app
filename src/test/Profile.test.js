import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/ConfigureStore';
import Profile from '../components/My_profile/Profile';

test('renders correctly', () => {
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <Profile />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
