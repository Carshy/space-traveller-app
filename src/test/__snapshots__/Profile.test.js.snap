// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="myProfile"
>
  <section
    className="my-comp"
  >
    <h2>
      My Mission
    </h2>
    <ul
      className="reserved-container"
    />
  </section>
  <section
    className="my-comp"
  >
    <h2>
      My Rockets
    </h2>
    <ul
      className="reserved-container"
    />
  </section>
</div>
`;
